### ISSUE TITLE
[QUERY] Migrating to new azure-messaging-eventhubs: SharedAccessSignatureTokenProvider & ConnectionStringBuilder missing

### ISSUE THREAD

**Issue by derSchtefan** [Azure/azure-sdk-for-java#24493]
**Query/Question**
After being informed by the IntelliJ plugin that the com.microsoft.azure::azure-eventhubs SDK is replaced by azure-messaging-eventhubs, we migrated, but saw that `SharedAccessSignatureTokenProvider` and `ConnectionStringBuilder` are missing. We use this in a cloud service to generate EventHub Publisher Policy bound tokens like described here (https://docs.microsoft.com/en-us/azure/event-hubs/event-hubs-features#publisher-policy).

The code used to look like this:

```java
val ttl = Duration.ofDays(tokenTimeToLive * 365L);
val authRule = this.getEventHubKey(eventHubConnectionStringConfig);

val sasToken = SharedAccessSignatureTokenProvider.generateSharedAccessSignature(
        eventHubConnectionStringConfig.getEventHubAuthorizationRuleName(), authRule.getKeys().primaryKey(), 
        URI.create("http://" + eventHubConnectionStringConfig.getEventHubUri() + "/" + eventHubConnectionStringConfig.getEventHubName()+ "/publishers/" + publisherName).toString(), ttl);

val connectionStringBuilder = new ConnectionStringBuilder().setEndpoint(eventHubConnectionStringConfig.getEventHubUri())
    .setNamespaceName(eventHubConnectionStringConfig.getEventHubNameSpace())
    .setEventHubName(eventHubConnectionStringConfig.getEventHubName() + "/publishers/" + publisherName)
    .setSharedAccessSignature(sasToken);

return connectionStringBuilder.toString();
```

This was quite useful and better than manually crafting the SAS token and connection string URL.
However, these two classes are gone -- are there replacement classes or mechanisms? Either in this or a management SDK?
Or do we have to manually reimplement the logic, following official Microsoft documentation?

***Why is this not a Bug or a feature Request?***
It is neither a bug nor a feature request because whoever removed the classes must have had a reason.

**Setup (please complete the following information if applicable):**
 - Library/Libraries: azure-messaging-eventhubs 5.10.1
 
 **Information Checklist**
 Kindly make sure that you have added all the following information above and checkoff the required fields otherwise we will treat the issuer as an incomplete report
- [X] Query Added
- [X] Setup information Added

**Comment by joshfree** [comment]
@conniey could you follow up with @derSchtefan on their question?

**Comment by conniey** [comment]
Hey, you would want to use [credential(String, String, AzureSasCredential)](https://azuresdkdocs.blob.core.windows.net/$web/java/azure-messaging-eventhubs/5.10.1/com/azure/messaging/eventhubs/EventHubClientBuilder.html#credential-java.lang.String-java.lang.String-com.azure.core.credential.AzureSasCredential-) when constructing your client from the EventHubClientBuilder. I hope this helps.

**Comment by ramya-rao-a** [comment]
@conniey Can you make sure our [migration guide](https://github.com/Azure/azure-sdk-for-java/blob/main/sdk/eventhubs/azure-messaging-eventhubs/migration-guide.md) covers this?

**Comment by ramya-rao-a** [comment]
Hey @derSchtefan 

Apologies for the late response here.

> It is neither a bug nor a feature request because whoever removed the classes must have had a reason.

You are right. The classes `SharedAccessSignatureTokenProvider` and `ConnectionStringBuilder` were removed for a reason. This was done to comply with the policy of not allowing connection strings that are hand built and not available via the Azure portal (or tooling like Azure CLI). There is along discussion in https://github.com/Azure/azure-sdk/issues/1701 where we concluded with adding an exception to this rule for the SharedAccessSignature i.e. the SAS token. The assumption was that one could follow the (docs for generating a signature(token) from a policy)[https://docs.microsoft.com/en-us/azure/event-hubs/authenticate-shared-access-signature#generating-a-signaturetoken-from-a-policy] and either build a connection string or use the builder that takes the sas token.

What we missed is the case of publisher policy which requires more support than just accepting the SAS token in the client builder. We will be taking this as a feature request for the next semester.
